text/0000-node-reactions.md

```diff
@@ -1,93 +1,69 @@
+Lightning Web Components(LWC) has support for similar lifecycle callbacks. Of particular interest to this RFC is the `connectedCallback` and `disconnectedCallback`. In LWC, the responsibility of invoking the callback is on the engine itself and this is inconvenient. Ideally the engine would delegate the work to a library or built in browser mechanism and just subscribe with its callbacks. The goal of this RFC is to abstract the responsibility of managing the lifecycle into a library and make LWC subscribe to lifecycle events from the node-reactions library. In the future, when LWC switches over to native web components, the transition to utilize the native web component callbacks mechanism will be straightforward.
 
-Further, there are inconsistencies in the current implementation. For example, connectedCallback should only be invoked if a custom element is connected to the document. In LWC, connectedCallback is invoked anytime a custom element is connected to a parent, regardless of wether the parent is connected to the document or not.
+Further, there are inconsistencies in the current implementation of callbacks in LWC engine. For example, connectedCallback should only be invoked if a custom element is connected to the document. In LWC, connectedCallback is invoked anytime a custom element is connected to a parent, regardless of whether the parent is connected to the document or otherwise.
 
-# Detailed design
+## Detailed design
 
-## When does a node qualify as connected?
+### When does a node qualify as connected?
 
-For custom elements, a node that is appended to the document or one of its connected discendents is considered as connected. [Node.isConnected](https://dom.spec.whatwg.org/#dom-node-isconnected) offers the most reliable way to determine if a node is connected.
+For custom elements, when an element is inserted into the document either directly or indirectly, it is considered as connected. [Node.isConnected](https://dom.spec.whatwg.org/#dom-node-isconnected) offers the most reliable way to determine if a node is connected.
 
-## Invariants
-1. A node can be connected and disconnected from the DOM more than once. Every time the event happens, the qualifying hook should be invoked.
-2. For a given node, more than one service can be registered to react to the same lifecycle event. In which case, each hook should be invoked in the order of registration(FIFO). 
-    2.1 Duplicate hooks for the same event type, for the same node will be ignored. For example, if a service registers the same connectedCallback more than once for the same node, the callback is invoked only once.
-3. Hooks are invoked synchronously.
-4. In case of an event affecting a subtree, nodes are processed in [preorder(or treeorder)](https://dom.spec.whatwg.org/#concept-tree-order).
+### Invariants
+1. A DOM element can be connected and disconnected from the DOM more than once. Every time the event happens, the qualifying hook should be invoked.
```

Comment:
hook -> callback

Reply:
there are still few places where you call hook instead of callback, can we get them all to be the same?